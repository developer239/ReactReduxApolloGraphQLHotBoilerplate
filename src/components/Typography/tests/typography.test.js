import React from 'react'
import { shallow } from 'enzyme'
import { theme } from 'styles'
import 'jest-styled-components'
import { H1, H2, H3, P, HR, StyledA, A } from '../index'


describe('Typography Components', () => {
  it('H1 has correct default styles', () => {
    const wrapper = shallow(
      <H1 theme={theme} />,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('H2 has correct default styles', () => {
    const wrapper = shallow(
      <H2 theme={theme} />,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('H3 has correct default styles', () => {
    const wrapper = shallow(
      <H3 theme={theme} />,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('P has correct default styles', () => {
    const wrapper = shallow(
      <P theme={theme} />,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('HR has correct default styles', () => {
    const wrapper = shallow(
      <HR theme={theme} />,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('StyledA has correct default styles', () => {
    const wrapper = shallow(
      <StyledA theme={theme} />,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('A has correct default styles', () => {
    const wrapper = shallow(
      <A theme={theme} />,
    )
    expect(wrapper).toMatchSnapshot()
  })
})
