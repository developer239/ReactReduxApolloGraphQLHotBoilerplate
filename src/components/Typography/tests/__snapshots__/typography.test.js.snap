// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography Components A has correct default styles 1`] = `
.c0 {
  color: #4a89dc;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #5d9cec;
}

.c0:active,
.c0:focus {
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<withProps(styled.a)
  theme={
    Object {
      "color": Object {
        "black": "#000",
        "darkGreen": "#37bc9b",
        "darkGrey": "#aab2bd",
        "darkRed": "#da4453",
        "lightGreen": "#48cfad",
        "lightGrey": "#ccd1d9",
        "lightRed": "#ed5565",
        "primary": "#5d9cec",
        "secondary": "#4a89dc",
        "white": "#fff",
      },
      "layout": Object {
        "maxWidth": "62.5rem",
      },
    }
  }
>
  <styled.a
    rel="noopener noreferrer"
    target="_blank"
    theme={
      Object {
        "color": Object {
          "black": "#000",
          "darkGreen": "#37bc9b",
          "darkGrey": "#aab2bd",
          "darkRed": "#da4453",
          "lightGreen": "#48cfad",
          "lightGrey": "#ccd1d9",
          "lightRed": "#ed5565",
          "primary": "#5d9cec",
          "secondary": "#4a89dc",
          "white": "#fff",
        },
        "layout": Object {
          "maxWidth": "62.5rem",
        },
      }
    }
  >
    <a
      className="c0"
      rel="noopener noreferrer"
      target="_blank"
    />
  </styled.a>
</withProps(styled.a)>
`;

exports[`Typography Components H1 has correct default styles 1`] = `
.c0 {
  font-weight: 500;
  margin: 0 0 24px 0;
  font-size: 21px;
  line-height: 24px;
}

.c0::selection {
  background: #5d9cec;
}

@media (min-width:34.375rem) {
  .c0 {
    font-size: 24px;
    line-height: 36px;
  }
}

@media (min-width:48rem) {
  .c0 {
    font-size: 36px;
    line-height: 48px;
  }
}

<styled.h1
  theme={
    Object {
      "color": Object {
        "black": "#000",
        "darkGreen": "#37bc9b",
        "darkGrey": "#aab2bd",
        "darkRed": "#da4453",
        "lightGreen": "#48cfad",
        "lightGrey": "#ccd1d9",
        "lightRed": "#ed5565",
        "primary": "#5d9cec",
        "secondary": "#4a89dc",
        "white": "#fff",
      },
      "layout": Object {
        "maxWidth": "62.5rem",
      },
    }
  }
>
  <h1
    className="c0"
  />
</styled.h1>
`;

exports[`Typography Components H2 has correct default styles 1`] = `
.c0 {
  font-weight: 500;
  margin: 0 0 24px 0;
  font-size: 18px;
  line-height: 24px;
}

.c0::selection {
  background: #5d9cec;
}

@media (min-width:34.375rem) {
  .c0 {
    font-size: 21px;
    line-height: 36px;
  }
}

@media (min-width:48rem) {
  .c0 {
    font-size: 24px;
    line-height: 24px;
  }
}

<styled.h2
  theme={
    Object {
      "color": Object {
        "black": "#000",
        "darkGreen": "#37bc9b",
        "darkGrey": "#aab2bd",
        "darkRed": "#da4453",
        "lightGreen": "#48cfad",
        "lightGrey": "#ccd1d9",
        "lightRed": "#ed5565",
        "primary": "#5d9cec",
        "secondary": "#4a89dc",
        "white": "#fff",
      },
      "layout": Object {
        "maxWidth": "62.5rem",
      },
    }
  }
>
  <h2
    className="c0"
  />
</styled.h2>
`;

exports[`Typography Components H3 has correct default styles 1`] = `
.c0 {
  font-weight: 500;
  margin: 0 0 24px 0;
  font-size: 16px;
  line-height: 24px;
}

.c0::selection {
  background: #5d9cec;
}

@media (min-width:34.375rem) {
  .c0 {
    font-size: 18px;
    line-height: 36px;
  }
}

@media (min-width:48rem) {
  .c0 {
    font-size: 21px;
    line-height: 24px;
  }
}

<styled.h3
  theme={
    Object {
      "color": Object {
        "black": "#000",
        "darkGreen": "#37bc9b",
        "darkGrey": "#aab2bd",
        "darkRed": "#da4453",
        "lightGreen": "#48cfad",
        "lightGrey": "#ccd1d9",
        "lightRed": "#ed5565",
        "primary": "#5d9cec",
        "secondary": "#4a89dc",
        "white": "#fff",
      },
      "layout": Object {
        "maxWidth": "62.5rem",
      },
    }
  }
>
  <h3
    className="c0"
  />
</styled.h3>
`;

exports[`Typography Components HR has correct default styles 1`] = `
.c0 {
  box-sizing: content-box;
  height: 0;
  display: block;
  unicode-bidi: isolate;
  overflow: hidden;
  margin-top: 25px;
  margin-bottom: 25px;
  border: 0;
  border-top: 1px solid #f2f2f4;
}

<styled.hr
  theme={
    Object {
      "color": Object {
        "black": "#000",
        "darkGreen": "#37bc9b",
        "darkGrey": "#aab2bd",
        "darkRed": "#da4453",
        "lightGreen": "#48cfad",
        "lightGrey": "#ccd1d9",
        "lightRed": "#ed5565",
        "primary": "#5d9cec",
        "secondary": "#4a89dc",
        "white": "#fff",
      },
      "layout": Object {
        "maxWidth": "62.5rem",
      },
    }
  }
>
  <hr
    className="c0"
  />
</styled.hr>
`;

exports[`Typography Components P has correct default styles 1`] = `
.c0 {
  font-weight: lighter;
  margin: 0 0 25px 0;
  font-size: 16px;
  line-height: 1.5;
}

.c0::selection {
  background: #5d9cec;
}

<styled.p
  theme={
    Object {
      "color": Object {
        "black": "#000",
        "darkGreen": "#37bc9b",
        "darkGrey": "#aab2bd",
        "darkRed": "#da4453",
        "lightGreen": "#48cfad",
        "lightGrey": "#ccd1d9",
        "lightRed": "#ed5565",
        "primary": "#5d9cec",
        "secondary": "#4a89dc",
        "white": "#fff",
      },
      "layout": Object {
        "maxWidth": "62.5rem",
      },
    }
  }
>
  <p
    className="c0"
  />
</styled.p>
`;

exports[`Typography Components StyledA has correct default styles 1`] = `
.c0 {
  color: #4a89dc;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #5d9cec;
}

.c0:active,
.c0:focus {
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<styled.a
  theme={
    Object {
      "color": Object {
        "black": "#000",
        "darkGreen": "#37bc9b",
        "darkGrey": "#aab2bd",
        "darkRed": "#da4453",
        "lightGreen": "#48cfad",
        "lightGrey": "#ccd1d9",
        "lightRed": "#ed5565",
        "primary": "#5d9cec",
        "secondary": "#4a89dc",
        "white": "#fff",
      },
      "layout": Object {
        "maxWidth": "62.5rem",
      },
    }
  }
>
  <a
    className="c0"
  />
</styled.a>
`;
